app/controllers/repos_controller.rb

```diff
@@ -41,13 +41,29 @@ def create
     @repo   = Repo.search_by(params[:repo][:name], params[:repo][:user_name]).first unless params_blank?
     @repo ||= Repo.new(repo_params)
     if @repo.save
-      @repo_sub = current_user.repo_subscriptions.create(repo: @repo)
-      flash[:notice] = "Added #{@repo.to_param} for triaging"
-      RepoSubscription.background_send_triage_email(@repo_sub.id)
-      redirect_to @repo
+      @repo_sub = current_user.repo_subscriptions.create(repo: @repo) if !RepoSubscription.find_by(user_id: current_user.id, repo_id: @repo.id)
+      
+      if @repo_sub
+        flash[:notice] = "Added #{@repo.to_param} for triaging"
+        RepoSubscription.background_send_triage_email(@repo_sub.id)
+        redirect_to @repo
+      else
+        flash[:error] = "Repository already exists"
+        @own_repos = GitHubBub.get("/user/repos", type: "owner", token: current_user.token).json_body
+        @starred_repos = GitHubBub.get("/user/starred", token: current_user.token).json_body
+        @watched_repos = GitHubBub.get("/user/subscriptions", token: current_user.token).json_body        
+        redirect_to new_repo_path
```

Comment:
I'll try to apply this code. Thanks for the suggestion! It's cool learning about that. :)